op-devstack/dsl/proofs/game_helper.go

```diff
@@ -0,0 +1,106 @@
+package proofs
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum-optimism/optimism/op-devstack/devtest"
+	"github.com/ethereum-optimism/optimism/op-devstack/dsl"
+	"github.com/ethereum-optimism/optimism/op-devstack/dsl/contract"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/wait"
+	"github.com/ethereum-optimism/optimism/op-service/txintent/bindings"
+	"github.com/ethereum-optimism/optimism/op-service/txplan"
+)
+
+type FaultDisputeGameHelper struct {
+	t       devtest.T
+	require *require.Assertions
+	game    *bindings.FaultDisputeGame
+}
+
+func NewFaultDisputeGameHelper(t devtest.T, require *require.Assertions, game *bindings.FaultDisputeGame) *FaultDisputeGameHelper {
+	return &FaultDisputeGameHelper{
+		t:       t,
+		require: require,
+		game:    game,
+	}
+}
+
+func (g *FaultDisputeGameHelper) GetMaxDepth() uint64 {
+	return contract.Read(g.game.MaxGameDepth()).Uint64()
+}
+
+func (g *FaultDisputeGameHelper) GetSplitDepth() uint64 {
+	return contract.Read(g.game.SplitDepth()).Uint64()
+}
+
+func (g *FaultDisputeGameHelper) GetRootClaim() *ClaimHelper {
+	return g.GetClaimAtIndex(int64(0))
+}
+
+func (g *FaultDisputeGameHelper) GetClaimAtIndex(claimIndex int64) *ClaimHelper {
+	claim := g.getClaimAtIndex(claimIndex)
+	return g.newClaimHelper(claimIndex, claim)
+}
+
+func (g *FaultDisputeGameHelper) Attack(eoa *dsl.EOA, claimIdx int64, newClaim common.Hash) {
+	claim := g.getClaimAtIndex(claimIdx)
+	g.t.Logf("Attacking claim %v (depth: %d) with counter-claim %v", claimIdx, claim.Position.Depth(), newClaim)
+
+	newPosition := claim.Position.Attack().ToGIndex()
+	requiredBond := contract.Read(g.game.GetRequiredBond(newPosition))
+
+	receipt := contract.Write(eoa, g.game.Attack(claim.Value, big.NewInt(claimIdx), newClaim), txplan.WithValue(requiredBond))
+	g.require.Equal(types.ReceiptStatusSuccessful, receipt.Status)
+}
+
+func (g *FaultDisputeGameHelper) newClaimHelper(claimIndex int64, claim bindings.Claim) *ClaimHelper {
+	return newClaimHelper(g.t, g.require, claimIndex, claim, g)
+}
+
+func (g *FaultDisputeGameHelper) getClaimAtIndex(claimIndex int64) bindings.Claim {
+	return contract.Read(g.game.ClaimData(big.NewInt(claimIndex))).Decode()
+}
+
+func (g *FaultDisputeGameHelper) getAllClaims() []bindings.Claim {
+	// TODO(#15948) - do we need to batch these? See: op-service/sources/batching.ReadArray
```

Comment:
batching would be nice. Getting all claims is fairly fast, but it can add up because it's a very common operation.